### DISCUSSION TITLE
How to completely change all Nodes/Edges 'in one go'

### DISCUSSION THREAD

**Discussion by royhuntley** [xyflow/xyflow#4315] [Q&A]
I have a reactflow instance that  is showing a collection of nodes and edges based upon the value of a dropdown box in another component.  
I dont seem to be able to work out how i can completely 'reset' the reactflow instance and replace all nodes/edges at once - is this something that is possible ?

**Comment by 7sStudio** [comment]
I have the same issue as well. When updating all nodes and edges in one go it causes the diagram to break, especially if the new nodes have similar ids to old ones. (in my case, the user clicks on a button that changes the entire diagram - while keeping the same reactflow instance)
